dimod/include/dimod/iterators.h

```diff
@@ -45,6 +45,23 @@ class NeighborhoodIterator {
         update_pair();
     }
 
+    NeighborhoodIterator(const NeighborhoodIterator<Bias, Index>& other)
```

Comment:
This looks good but declaring an explicit copy constructor may invalidate the implicit destructor and move constructor. The rules are listed here, it is not absolutely needed to define the move constructor if it is deleted due to the declaration of explicit copy constructor, but we may want to do that to improve speed. And the destructor is needed to avoid memory leaks. 

Happy reading :) 
https://stackoverflow.com/questions/46402983/rule-of-five-and-implicitly-deleted-functions